tests/client/test_client.py

```diff
@@ -1,7 +1,34 @@
 import unittest
 from unittest.mock import MagicMock, patch
 
+import pytest
+import requests
+import responses
+
 from caselawclient.Client import MarklogicApiClient
+from caselawclient.errors import GatewayTimeoutError
+
+
+class TestErrors(unittest.TestCase):
+    def setUp(self):
+        self.client = MarklogicApiClient("", "", "", False)
+
+    def test_timeout(self):
+        with responses.RequestsMock() as response_list:  # type: ignore[attr-defined]
+            response_list.add(
+                responses.GET,  # type: ignore[attr-defined]
+                url="http://example.com",
+                status=504,
+                body="Example Gateway Timeout.",
+            )
+            response = requests.get("http://example.com")
+
+        # response = MagicMock()
+        # response.raise_for_status.side_effect=HTTPError(response=response)
+        # response.status_code=504
+        with pytest.raises(GatewayTimeoutError) as gateway_exception:
+            self.client._raise_for_status(response)
```

Comment:
Not for now, but this got me thinking, maybe we could create a `MarklogicResponse` class that looks like

````
class MarkLogicResponse(requests.Response):
    def raise_for_status(self):
        # .... current _raise_for_status implementation here ....
```

that way we would separate concerns a bit and thus no longer need to test a private method